copy/copy.go

```diff
@@ -262,14 +277,14 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 		manifestUpdates: &types.ManifestUpdateOptions{InformationOnly: types.ManifestUpdateInformation{Destination: c.dest}},
 		src:             src,
 		// diffIDsAreNeeded is computed later
-		canModifyManifest: len(sigs) == 0,
+		canModifyManifest: len(sigs) == 0 && !destIsDigestedReference,
 		// Ensure _this_ copy sees exactly the intended data when either processing a signed image or signing it.
 		// This may be too conservative, but for now, better safe than sorry, _especially_ on the SignBy path:
 		// The signature makes the content non-repudiable, so it very much matters that the signature is made over exactly what the user intended.
 		// We do intend the RecordDigestUncompressedPair calls to only work with reliable data, but at least there’s a risk
 		// that the compressed version coming from a third party may be designed to attack some other decompressor implementation,
 		// and we would reuse and sign it.
-		canSubstituteBlobs: len(sigs) == 0 && options.SignBy == "",
+		canSubstituteBlobs: len(sigs) == 0 && options.SignBy == "" && !destIsDigestedReference,
```

Comment:
(Hum, at this point it might be more maintainable to have a separate
```go
ic.canSubstituteBlobs = ic.canModifyManifest && options.SignBy == ""
```
)

Reply:
Made the change as you described.